maestrowf/datastructures/environment/gitdependency.py

```diff
@@ -146,21 +147,30 @@ def acquire(self, substitutions=None):
                 self.path = substitution.substitute(self.path)
                 self.url = substitution.substitute(self.url)
 
+        # Check that we are connected to the internet.
+        logger.debug("Pinging Google to verify internet connectivity...")
+        ping_url("https://www.google.com")
```

Comment:
Does a failed connection automatically cancel the workflow?

Reply:
Yeah, if `ping_url` can't find the URL it throws an exception. We acquire before starting the workflow, so it'd not even get to launching the conductor.

Reply:
I've changed this to check the repo url directly.